sqs_workers/async_task.py

```diff
@@ -30,8 +30,11 @@ def run(self, *args, **kwargs):
         """
         Run the task synchronously.
         """
-        kwargs = bind_arguments(self.processor, args, kwargs)
-        return self.processor(**kwargs)
+        if self.queue.batching_policy.batching_enabled:
+            return self.processor([kwargs])
+        else:
+            kwargs = bind_arguments(self.processor, args, kwargs)
+            return self.processor(**kwargs)
```

Comment:
Something seems wrong here 🤔

- Without batching: we use `bind_arguments()` to convert `(*args, **kwargs)` to a single `kwargs` dict, which we can then pass to the task function as `function(**kwargs)`
- With batching: we pass raw kwargs... and just ignore positional arguments!?

Maybe that's what we want. But in that case, I'd prefer to fail loudly if `args` is not empty, so that developers immediately realize that.

Reply:
Maybe we should even use something like `kwargs = bind_arguments(self.processor, [[kwargs]], {})` if batching is enabled, to also fail loudly if `self.processor` doesn't accept a single positional argument.

Reply:
_Rubber ducking this, a bit stream of conscious I'm afraid:_

In our readme we're clear that for non batched read messages kwargs should be used, as they are matched to the args in the call handler:

```python
@queue.processor("send_email")
def send_email(to, subject, body):
    print(f"Sending email {subject} to {to}")

send_email.delay(to="user@example.com", subject="Hello world", body="hello world")
```

Playing with `bind_arguments` this works, so refactored to this:

```python
kwargs = bind_arguments(self.processor, [[kwargs]], {})
return self.processor(**kwargs)
```

However in this instance we are still throwing away the args.

Because for batching we can't infer the parameter name from the underlying call handler, we wouldn't be able to map any args into kwargs based on position (which is what I believe `bind_arguments` is doing currently?)

So I've done as suggested and raising an error if `args` isn't empty for a batch task.

```python
if self.queue.batching_policy.batching_enabled:
    if len(args) > 0:
        raise TypeError("Must use keyword arguments only for batch read queues")
    kwargs = bind_arguments(self.processor, [[kwargs]], {})
    return self.processor(**kwargs)
else:
    kwargs = bind_arguments(self.processor, args, kwargs)
    return self.processor(**kwargs)
```